ita-social-projects/StreetCode_Client
# Dynamics meta tags

**Frontend server for handling dynamic meta tags for news and streetcode**

**Steps to setup:**
- open terminal in project root folder (StreetCode_Client)
- run "npm run build" command and wait for dist folder to generate
- open dist folder and create file "env-config.js"
- paste this data in file WITH UNDERSCORES AROUND env (like on image): window._env_ = { REACT_APP_BACKEND_URL: "https://localhost:5001/api",  REACT_APP_TEMPVAL: "tempval",  API_URL: "https://stageback.streetcode.com.ua/api",  SERVER_API_URL: "http://localhost:4000",  REACT_APP_GOOGLE_ANALYTICS: "G-EBJQX1FFF0", }
<img width="534" alt="Screenshot 2024-01-09 at 7 21 03 PM" src="https://github.com/ita-social-projects/StreetCode_Client/assets/91199925/b213ebdc-a92f-4213-a703-5652ce82883c">

**Steps to start:**
- open terminal in server folder (StreetCode_Client/server)
- run "npm run start" command
- visit localhost:4000, you should see streetcode main page

** For now server uses stageback for API, not a localhost. So be careful with your action in admin-panel **

**How to test:**
- open terminal in server folder (StreetCode_Client/server)
- run "npm run test" -> you should see 22 tests passed
- run "npm run test:cov" -> you should see code coverage >80% for All files Stmts
- run "npm run lint" -> try to commit, no files should be changed

**How to test logic:**
- open google chrome and add the extention https://chromewebstore.google.com/detail/social-share-preview/ggnikicjfklimmffbkhknndafpdlabib
- run server -> open website -> run installed extention
- visit streetcode page -> in extention card you should see "ч/б портрет, назва стріткоду, «Стріткод: історія на кожному кроці»"
- visit news page -> in extention card you should see "картинка з новини і її заголовок."
- visit main page or any other page exept streetcode or news  ->  in extention card you should see "перший банер з
головної, «Стріткод: історія на кожному кроці» "

- try to edit some **news** and refresh the news page -> you should see updated data in extention card
- try to edit some **streetcode** and refresh the streetcode page -> you should see updated data in extention card
- create new **news** and visit the news page -> in extention card you should see the news data
- create new **streetcode** and visit the streetcode page -> in extention card you should see the news data
- delete some **news** and visit the news url -> you should see 404 page and the same  extention card info as on any other page: banner + :«Стріткод: історія на кожному кроці» "
- delete some **streetcode** and visit the streetcode url -> you should see 404 page and the same  extention card info as on any other page: banner + :«Стріткод: історія на кожному кроці» "

**That's all folks, thanks 👍**
